_festival.cpp

```diff
@@ -46,25 +46,34 @@ static char _textToWav_doc[] = "_textToWav(text) -> unicode filename\n"
     "This is a private method.";
 static PyObject* _textToWav(PyObject* self, PyObject* args) {
     const char* text;
-    if (!PyArg_ParseTuple(args, "s:_textToWav", &text)) return NULL;
-    
+    const char* saved_filename;
+
+    EST_String tmpfile;
+    int argc = PyTuple_GET_SIZE(args);
+    if (argc == 1){
+        if (!PyArg_ParseTuple(args, "s:_textToWav", &text)) return NULL;
+        tmpfile = make_tmp_filename();
+    } else {
+        if (!PyArg_ParseTuple(args, "s|s:_textToWav", &text, &saved_filename)) return NULL;
```

Comment:
Since we're accepting a user-inputted filepath here, it might be wise to do some checks to make sure the parent folder exists and is writeable, and fail gracefully if not. (This could be done up one level in the python `textToWav()` method).